tachyon/math/finite_fields/fp3.h

```diff
@@ -27,7 +29,65 @@ class Fp3 final : public CubicExtensionField<Fp3<Config>> {
 
   constexpr static uint64_t kDegreeOverBasePrimeField = 3;
 
-  static void Init() { Config::Init(); }
+  static void Init() {
+    Config::Init();
+    // x³ = q = Config::kNonResidue
+
+    // αᴾ = (α₀ + α₁x + α₂x²)ᴾ
+    //    = α₀ᴾ + α₁ᴾxᴾ + α₂ᴾx²ᴾ
+    //    = α₀ + α₁xᴾ + α₂x²ᴾ <- Fermat's little theorem
+    //    = α₀ + α₁xᴾ⁻¹x + α₂x²ᴾ⁻²x²
+    //    = α₀ + α₁(x³)^((P - 1) / 3) * x + α₂(x³)^(2 * (P - 1) / 3) * x²
+    //    = α₀ + α₁ωx + α₂ω²x², where ω is a cubic root of unity.
+
+    constexpr uint64_t N = BasePrimeField::kLimbNums;
+    // m₁ = P
+    mpz_class m1;
+    gmp::WriteLimbs(BasePrimeField::Config::kModulus.limbs, N, &m1);
+
+#define SET_M(d, d_prev) mpz_class m##d = m##d_prev * m1
+
+    // m₂ = m₁ * P = P²
+    SET_M(2, 1);
+
+#undef SET_M
+
+#define SET_EXP_GMP(d) mpz_class exp##d##_gmp = (m##d - 1) / mpz_class(3)
+
+    // exp₁ = (m₁ - 1) / 3 = (P¹ - 1) / 3
+    SET_EXP_GMP(1);
+    // exp₂ = (m₂ - 1) / 3 = (P² - 1) / 3
+    SET_EXP_GMP(2);
+
+#undef SET_EXP_GMP
+
+    // kFrobeniusCoeffs[0] = q^((P⁰ - 1) / 3) = 1
+    Config::kFrobeniusCoeffs[0] = FrobeniusCoefficient::One();
+#define SET_FROBENIUS_COEFF(d)                \
+  BigInt<d * N> exp##d;                       \
+  gmp::CopyLimbs(exp##d##_gmp, exp##d.limbs); \
+  Config::kFrobeniusCoeffs[d] = Config::kNonResidue.Pow(exp##d)
+
+    // kFrobeniusCoeffs[1] = q^(exp₁) = q^((P¹ - 1) / 3) = ω
+    SET_FROBENIUS_COEFF(1);
+    // kFrobeniusCoeffs[2] = q^(exp₂) = q^((P² - 1) / 3)
+    SET_FROBENIUS_COEFF(2);
+
+#undef SET_FROBENIUS_COEFF
+
+    // kFrobeniusCoeffs[0] = q^(2 * (P⁰ - 1) / 3) = 1
```

Comment:
```suggestion
    // kFrobeniusCoeffs[0] = q^((P⁰ - 1) / 3) = 1
```

Reply:
I think this should be

```suggestion
    // kFrobeniusCoeffs2[0] = q^(2 * (P⁰ - 1) / 3) = 1

```

Reply:
@chokobole Oh then should https://github.com/kroma-network/tachyon/pull/101/files#diff-e673446d9e3e55b20b8c313a638d62f9e9af8b2135d6dd4745d5e3ca601fadf1R182 be fixed with 2 * at the front?

Reply:
You're right! I'll fix it.